pandas/core/algorithms.py

```diff
@@ -989,6 +989,11 @@ def _get_data_algo(values, func_map):
         f = func_map['uint64']
         values = _ensure_uint64(values)
 
+    elif is_categorical_dtype(values) and values.ordered:
+        nanMapper = np.vectorize(lambda t: np.NaN if t == -1 else t*1.)
```

Comment:
@ikilledthecat You can do this vectorized, which will be much more performant than applying this function on each element (like `codes[codes==-1] = np.nan`, after converting it to float)